modules/src/ics04_channel/handler/recv_packet.rs

```diff
@@ -78,7 +79,7 @@ pub fn process(ctx: &dyn ChannelReader, msg: MsgRecvPacket) -> HandlerResult<Pac
 
     // Check if packet timestamp is newer than the local host chain timestamp
     let latest_timestamp = ctx.host_timestamp();
-    if (packet.timeout_timestamp != 0) && (packet.timeout_timestamp <= latest_timestamp) {
+    if let Expiry::Expired = latest_timestamp.check_expiry(&packet.timeout_timestamp) {
```

Comment:
I am not too sure why there are different comparisons using `<=` or `>` in different places. I try to keep them consistent by always using `>` inside `check_expiry`. It is also not clear whether we need a strict equality here, considering this requires the exact same nanoseconds to satisfy.

Reply:
Where is the equality used ?

Reply:
The replacement with check_expiry are consistent. The >= or < are used interchanged, following the time flow intuition,   and there is an equivalent representation with just one, for which you picked a good name :).

Reply:
The original comparator is `packet.timeout_timestamp <= latest_timestamp`. If we strictly switch the position, it becomes `latest_timestamp >= packet.timeout_timestamp`. But in the new code I am doing `latest_timestamp > packet.timeout_timestamp`.